cooperator/models/subscription_request.py

```diff
@@ -256,10 +257,11 @@ def _compute_subscription_amount(self):
     share_product_id = fields.Many2one(
         "product.product",
         string="Share type",
-        domain=[("is_share", "=", True)],
+        domain="[('is_share', '=', True), ('company_id', 'in', (company_id, False))]",
         required=True,
         readonly=True,
         states={"draft": [("readonly", False)]},
+        check_company=True,
```

Comment:
`check_company` and `domain` are not related. `check_company` is a safeguard provided by odoo (called in `.write()` and `.create()`) to avoid creating inconsistent data. the domain is only applied in the ui, while this will prevent code to link to shares of other companies.